ReactiveTask/Task.swift

```diff
@@ -154,39 +154,42 @@ private final class Pipe {
 	/// anywhere else, as it may close unexpectedly.
 	func transferReadsToProducer() -> ReadProducer {
 		return SignalProducer { observer, disposable in
-			dispatch_group_enter(self.group)
-			let channel = dispatch_io_create(DISPATCH_IO_STREAM, self.readFD, self.queue) { error in
+			self.group.enter()
+			let channel = DispatchIO(type: .stream, fileDescriptor: self.readFD, queue: self.queue) { error in
 				if error == 0 {
 					observer.sendCompleted()
 				} else if error == ECANCELED {
 					observer.sendInterrupted()
 				} else {
-					observer.sendFailed(.POSIXError(error))
+					observer.sendFailed(.posixError(error))
 				}
 
 				close(self.readFD)
-				dispatch_group_leave(self.group)
+				self.group.leave()
 			}
 
-			dispatch_io_set_low_water(channel, 1)
-			dispatch_io_read(channel, 0, Int.max, self.queue) { (done, data, error) in
-				if let data = data {
-					observer.sendNext(data as! NSData)
+			channel.setLimit(lowWater: 1)
+			channel.read(offset: 0, length: Int.max, queue: self.queue) { (done, dispatchData, error) in
+				if let dispatchData = dispatchData {
+					let bytes = UnsafeMutablePointer<UInt8>.allocate(capacity: dispatchData.count)
+					dispatchData.copyBytes(to: bytes, count: dispatchData.count)
+					let data = Data(bytes: bytes, count: dispatchData.count)
```

Comment:
We should create this data like the other `DispatchData -> NSData`. Maybe it's worth adding a convenience init to `NSData` that takes a `DispatchData`?